src/librustc_trans/mir/block.rs

```diff
@@ -334,14 +334,12 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
                         let len = self.trans_operand(&mut bcx, len).immediate();
                         let index = self.trans_operand(&mut bcx, index).immediate();
 
-                        let const_err = common::const_to_opt_uint(len).and_then(|len| {
-                            common::const_to_opt_uint(index).map(|index| {
-                                ErrKind::IndexOutOfBounds {
-                                    len: len,
-                                    index: index
-                                }
-                            })
-                        });
+                        let const_err = common::const_to_opt_u128(len, false)
+                            .and_then(|len| common::const_to_opt_u128(index, false)
+                                .map(|index| ErrKind::IndexOutOfBounds {
```

Comment:
Hmm, the function was removed because it would cause bad behaviour in cross compilation situations...

I guess I'll add a const_to_opt_u64 function instead, as there are no 128 bit targets nor hosts neither in existence nor on the horizon.